File: CodeWars/findNthPrime.py
# Problem: Given any n N find the N'th prime.
# Solution: 

from math import log

def sieve_of_eratosthenes(n):
    prime = [True for num in range(n + 1)]
    p = 2
    while (p * p <=n):
        if (prime[p]):
            for i in range(p * p, n + 1, p):
                prime[i] = False

        p += 1
    primes = []
    for p in range(2, n + 1):
        if prime[p]:
            primes.append(p)
    return primes

# Find the nth prime. 
def find_nth_prime(n):
    # Estimate for the upper limit of the n-th prime
    if n < 6:
        limit = 15  # Small n values adjustment
    else:
        limit = int(n * (log(n) + log(log(n))))  # n log n for larger values

    primes = sieve_of_eratosthenes(limit)
    if n <= len(primes):
        return primes[n-1]
    else:
        return None  # n is too large for the estimated limit

File: CodeWars/findNthPrime.js
// Given any N find the Nth prime.

function sieveOfEratosthenes(limit) {
    let primes = new Array(limit + 1).fill(true);
    primes[0] = false;
    primes[1] = false;
    const sqrtLimit = Math.sqrt(limit);

    for (let p = 2; p <= sqrtLimit; p++) {
        if (primes[p]) {
            for (let i = p * p; i <= limit; i += p) {
                primes[i] = false;
            }
        }
    }

    let primeNumbers = [];
    for (let p = 2; p <= limit; p++) {
        if (primes[p]) primeNumbers.push(p);
    }
    return primeNumbers;
}

function findNthPrime(n) {
    let limit;
    if (n < 6) {
        limit = 15; // A small initial limit for small n values
    } else {
        limit = Math.ceil(n * (Math.log(n) + Math.log(Math.log(n)))); // n log n for larger values
    }

    let primes = sieveOfEratosthenes(limit);
    if (n <= primes.length) {
        return primes[n - 1];
    } else {
        return null; // n is too large for the estimated limit
    }
}